exercises/advriskmin/ex_rnw/sol_ic_advriskmin_1.Rnw

```diff
@@ -0,0 +1,87 @@
+
+\begin{enumerate}
+  \item 
+  
+  	\lz
+    	
+  	As seen in \href{https://github.com/slds-lmu/lecture_i2ml/raw/master/slides-pdf/slides-advriskmin-regression-l2.pdf}{L2-loss slide 2}, the risk minimizer $\fbayes$ is the conditional mean:
+    	
+    	
+  	$$  \fxbayes = \E_{y_{|x}} \left[ y_{|x} \right] $$
+    	
+    \lz 
+  	The distribution of $y$ given $x$ is known, and can be plugged-in:
+    	
+  	$$ \fxbayes = \E\left[ \normal(a+bx,1) \right] = a+bx$$
+  	
+  	\lz 
+  
+  
+  \item 
+  
+  	The resulting risk can be calculated by using the definition:
+  	
+  	\begin{align*}
+  	  \risk_L(\fbayes)  &= \E_{xy} \left[ (y-\fxbayes)^2 \right] \\
+  	  &= \E_{xy} \left[ (y-\E_{y_{|x}} \left[ y_{|x} \right])^2 \right] \\
+  	  &= \E_x \left[ \var_{y_{|x}} (y_{|x}) \right]  \\ 
+  	  &= \E_x \left[ \var (\normal(a+bx,1)) \right] \\
+  	  &= \E_x \left[ 1 \right] \\
+  	  &= 1
+  	\end{align*}
+  	 
+   	
+  \lz
+  
+  \item
+    
+    The risk minimizer for the L2 loss is the conditional mean. Considering that the hypothesis space is now restricted to constant models, $f(x)$ is a constant for any $x$. The optimal constant model in terms of the theoretical risk for the L2 loss is the expected value over $y$.
+    \lz
+    $$
+    \bar{f}(x) = \E_{y_{|x}} \left[ y_{|x} \right] = \E_y[y]
+    $$
+    
+    The \href{https://github.com/slds-lmu/lecture_i2ml/blob/master/slides/mathrefresher/slides-probability-theory.pdf}{Law of total expectation} can be used in this case:
+    \lz
+    \begin{align*}
+    \E_y[y] &= \E_x \left [ \E_{y_{|x}} \left[ y_{|x} \right] \right] \\
+    &= \E_x \left[ a+ bx \right] \\ 
+    &= a + b \cdot \E_x[x] \\
+    &= a + b \cdot 0 \\
+    &= a 
+    \end{align*}
+    
+    \lz
+  
+  \item 
+    
+    To calculate the risk of the optimal constant model, the definition can be used: 
+    \lz
+    \begin{align*}
+      \risk_L(\bar{f}) &= \E_{xy} \left[ (y - \bar{f}(x) )^2 \right]\\
+      &= \E_{xy} \left[ (y - a)^2 \right]\\
+      &= \E_y [ y^2 ] - 2a \underbrace{\E_y[y]}_{a} + a^2 \\
+      &= \E_y [ y^2 ] - a^2 
+    \end{align*}
+    
+    As $\E_y[y^2]$ is yet unknown, the calculation is detailed below: 
+    \lz
+    \begin{align*} 
+      \E_y [y^2] &= \E_x \left[ \E_{y_{|x}} \left[ y^2_{|x} \right] \right] \\
+      &= \E_x \left[ \var_{y_{|x}} [y_{|x}] + \E_{y_{|x}} \left[ y_{|x} \right] \right] \\
```

Comment:
I guess, here is a square missing at the end (squaring the E?)

Reply:
Indeed! I changed it in the next commit

Reply:
> Could you rename the files, please, @patriciomassaro ? It should be "ic_sol_...". Thank you

Done